SameCodon.pm

```diff
@@ -84,64 +84,23 @@ sub run {
     return {} if grep {!$_->isa('Bio::EnsEMBL::Mapper::Coordinate')} @coords;
     
     my @results;
-    
     # we might get multiple "slices" if the codon that the variant falls in spans exons
     foreach my $coord(@coords) {
         
         my ($slice_start, $slice_end) = ($coord->start, $coord->end);
         
-        # using cache?
-        if(defined($config->{cache})) {
-            
-            # spoof region based on cache region size
-            my $size = $config->{cache_region_size};
-            
-            my $s = (int ($vf_start / $size) * $size) + 1;
-            my $e = (int ($vf_end / $size) + 1) * $size;
-            my $c = $vf->{chr};
-            
-            my $region = "$s\-$e";
-            
-            my $vf_cache = $self->{_cache}->{$c}->{$region} || load_dumped_variation_cache($config, $c, $region);
-            
-            foreach my $pos($slice_start..$slice_end) {
-                if(my $existing_vars = $vf_cache->{$c}->{$pos}) {
-                    push @results,
-                        map {$_->{variation_name}}
-                        grep {
-                            $_->{variation_name} ne $vf->variation_name &&
-                            $_->{failed} <= $config->{failed} &&
-                            $_->{start} != $vf_start &&
-                            $_->{end} != $vf_end &&
-                            scalar $mapper->genomic2cds($_->{start}, $_->{end}, 1) >= 1
-                        }
-                        @$existing_vars;
-                }
+        my $sub_slice = $vf->slice->sub_Slice($slice_start, $slice_end);
+        my $vf_adaptor = $vf->slice->_get_VariationFeatureAdaptor();
+        my @test = $vf_adaptor->fetch_all_by_Slice_SO_terms($sub_slice);
```

Comment:
Is this a bit of leftover debug code?

Reply:
No, it just (very hackily) enabled this plugin to check the cache for variants instead of having to use the DB.

There's no nice way of doing this in new VEP, so probably best to ditch it.

If you want this functionality, it would probably have to go in the main code.

Reply:
Hi @willmclaren !   
The functionality has been requested. Yes - it would be better to add it to the main code, though this probably isn't as quick. I guess you are concerned it will be slower going to the cache a second time?

Reply:
In this implementation here it was going to the cache a second time, so not especially fast.

The problem is that plugins don't (currently!) have access to the AnnotationSource objects that you'd need to go back to the cache.

If you implemented this in the main code you've still got a few gotchas. When known/overlapping variants are loaded (in the AnnotationSource code) VEP only has genomic positions; transcript mappings are not pulled in until you reach the OutputFactory stage, by which time it is assumed you've already retrieved everything you needed from each AnnotationSource. A quick hack would be to cache everything 3bp up- and downstream, but then what if your codon spans more than one exon (presumably this happens?).

IIRC you can't enforce the transcript AnnotationSources loading first (ie to get the transcript mappings to then use when loading known variants) as in fact the variant AnnotationSource is given priority as it can filter variants (the idea being that filtered variants then aren't wasting cycles getting annotated).

TLDR: making the plugin use the DB is probably the easiest approach for now!